core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -63,7 +63,6 @@ object RequestChannel extends Logging {
     // o.a.k.common.requests.AbstractRequest.getRequest()
     private val keyToNameAndDeserializerMap: Map[Short, (ByteBuffer) => RequestOrResponse]=
       Map(ApiKeys.FETCH.id -> FetchRequest.readFrom,
-        ApiKeys.METADATA.id -> TopicMetadataRequest.readFrom,
```

Comment:
Is this another compatibility gray area? I was thinking we could just leave the whole TopicMetadataRequest around until we delete it. I'm fine removing it now too if the concern is minimal.